packages/studio/src/index.ts

```diff
@@ -260,6 +280,12 @@ export type PanelExtensionContext = {
    */
   readonly dataSourceProfile?: string;
 
+  /**
+   * An array of metadata entries. Each entry includes a name and a map of key-value pairs
+   * representing the metadata associated with that name (only avaiable in MCAP files).
+   */
+  readonly metadata: ReadonlyArray<Readonly<Metadata>>;
```

Comment:
Why it has to be ReadonlyArray<Readonly<Metadata>>?
Readonly<Metadata> wouldn't be enough?

Reply:
Because the internal objects inside the array should be immutable as well.

If you only use ReadonlyArray<Metadata>, you could't change the array, but it would be possible to do something like: metadata[0].name = "other name", because this object would not be Readonly.

Note: This is for Typescript only, the object is deeply frozen anyway, so even if some developer try to use @ts-ignore, it will not be possible to modify it.